R/Colombia.R

```diff
@@ -21,64 +21,108 @@ Colombia <- R6::R6Class("Colombia",
     #' @field origin name of origin to fetch data for
     origin = "Colombia",
     #' @field supported_levels A list of supported levels.
-    supported_levels = list("1"),
+    supported_levels = list("1", "2"),
     #' @field supported_region_names A list of region names in order of level.
-    supported_region_names = list("1" = "departamento"),
+    supported_region_names = list(
+      "1" = "departamento",
+      "2" = "municipio"
+    ),
     #' @field supported_region_codes A list of region codes in order of level.
-    supported_region_codes = list("1" = "iso_3166_2"),
+    supported_region_codes = list(
+      "1" = "iso_3166_2",
+      "2" = "codigo_municipio"
+    ),
     #' @field common_data_urls List of named links to raw data.
     # nolint start
     common_data_urls = list(
-      "main" = "https://raw.githubusercontent.com/danielcs88/colombia_covid-19/master/datos/cronologia.csv"
+      "main" = "https://www.datos.gov.co/resource/gt2j-8ykr.csv?$select=fecha_diagnostico,ciudad_municipio"
     ),
     # nolint end
     #' @field source_data_cols existing columns within the raw data
-    source_data_cols = c("cases_total"),
+    source_data_cols = c("cases_new"),
     #' @field source_text Plain text description of the source of the data
-    source_text = "Daniel C\u00e1rdenas",
+    source_text = "Datos abiertos Colombia (Colombia open data)",
     #' @field source_url Website address for explanation/introduction of the
     #' data
-    source_url = "https://github.com/danielcs88/colombia_covid-19/",
+    source_url = "https://www.datos.gov.co/Salud-y-Protecci-n-Social/Casos-positivos-de-COVID-19-en-Colombia/gt2j-8ykr", # nolint
 
     #' @description Set up a table of region codes for clean data
     #' @importFrom tibble tibble
     #' @importFrom dplyr mutate
     set_region_codes = function() {
-      self$codes_lookup$`1` <- covidregionaldata::colombia_codes
+      self$codes_lookup$`1` <- covidregionaldata::colombia_codes %>%
+        select(level_1_region, level_1_region_code) %>%
+        unique()
+      self$codes_lookup$`2` <- covidregionaldata::colombia_codes
+    },
+
+    #' @description Colombia specific download using Socrata API
+    #' This uses the `RSocrata` package if it is installed or downloads
+    #' a much larger csv file if that package is not available.
+    #' @importFrom dplyr select
+    download = function() {
+      message_verbose(self$verbose,
+                      "Downloading Colombia data. This may take a while.")
+      # RSocrata package is recommended but not required
+      if (require(RSocrata, attach.required = TRUE)) {
+        self$data$raw$main <- RSocrata::read.socrata(self$data_urls[["main"]])
+      } else {
+        # If the RSocrata package is not available, we download the full wide
```

Comment:
I like this but do we need a workaround? Could we instead just expect people to install `RScrata` if they want this data? The error message from `requireNamespace` should be informative already + can put a flag in the docs.

Reply:
(are there other advantages to the csv route?)

Reply:
I am *mixed* on the work-around. I think that having a fall-through that works is good, in case `RSocrata` suddenly evaporates (though honestly, it's been around longer than this package) or in case someone has a particular reason not to want to use the Socrata API (they cannot install the package for technical or legal reasons, perhaps). 

That said, it's another six lines of code which could break which we have to keep around (and notionally test and maintain). If you prefer we drop the work-around I can just make this a `stop`.

Reply:
I think my preference would be to try and keep maintenance down but I don't mind either way and it is great to have a fallback.

Reply:
I've removed the workaround and put in a `stop` if the `download` method is used without `RSocrata` installed.